### DISCUSSION TITLE
Feature request: allow downloading of generated TTS

### DISCUSSION THREAD

**Discussion by cduk** [open-webui/open-webui#5001] [Ideas]
Currently text to speech is supported and the output is cached in fragments.

It would be useful if the UI presented an option to download the generated speech as a single MP3 file.

**Comment by silentoplayz** [comment]
There is an ElevenLabs TTS Function available on the Open WebUI community website that you can import directly into your Open WebUI instance. Once configured, you can directly download audio of generated speech from text produced via Text-To-Speech.

https://openwebui.com/f/justinrahb/elevenlabs_tts/

**Reply by verynewtocode** [reply]
why do i have to sign up to download

**Reply by silentoplayz** [reply]
> why do i have to sign up to download

You don't have to if you don't desire to, but I will mention that the community website hosts community made and shared workspace model templates, prompts, tools, and functions for signed up users to browse and download from other users.

I've tried importing this very same `ElevenLabs TTS` function into my updated Open WebUI instance today and it failed to successfully import due to a missing module. It's likely nothing that following along with https://docs.openwebui.com/features/plugin/migration/ couldn't fix.
<img width="440" height="178" alt="image" src="https://github.com/user-attachments/assets/05fe8e2d-88c9-4cb0-8893-f7c95dac45c6" />

**Comment by cduk** [comment]
Thanks. Does this work only with ElevenLabs generated TTS samples? I currently use an OpenAI compatible audio endpoint which points to a custom server.

**Reply by silentoplayz** [reply]
The ElevenLabs TTS Function code is specifically designed to work with ElevenLabs Text-to-Speech (TTS) API. However, you are free to modify it to work with your custom TTS server by replacing the requests to the ElevenLabs API with requests to your custom server. The code is licensed under the MIT License, which allows you to freely use, modify, and distribute the code for any purpose, including in commercial or proprietary projects.
